span_test.go

```diff
@@ -105,25 +105,40 @@ func TestSpanOperationName(t *testing.T) {
 }
 
 func TestSetTag_SamplingPriority(t *testing.T) {
-	tracer, closer := NewTracer("DOOP", NewConstSampler(true), NewNullReporter(),
-		TracerOptions.DebugThrottler(throttler.DefaultThrottler{}))
-
-	sp1 := tracer.StartSpan("s1").(*Span)
-	ext.SamplingPriority.Set(sp1, 0)
-	assert.False(t, sp1.context.IsDebug())
-
-	ext.SamplingPriority.Set(sp1, 1)
-	assert.True(t, sp1.context.IsDebug())
-	assert.NotNil(t, findDomainTag(sp1, "sampling.priority"), "sampling.priority tag should be added")
-	closer.Close()
+	type testCase struct {
+		noDebugFlagOnForcedSampling bool
+		throttler                   throttler.Throttler
+		samplingPriority            uint16
+		expDebugSpan                bool
+		expSamplingTag              bool
+	}
 
-	tracer, closer = NewTracer("DOOP", NewConstSampler(true), NewNullReporter(),
-		TracerOptions.DebugThrottler(testThrottler{allowAll: false}))
-	defer closer.Close()
+	testSuite := []testCase{
+		{false, throttler.DefaultThrottler{}, 0, false, false},
```

Comment:
nit: I would prefer we used a named struct, as show below because it is easier to keep track of the variables. 
```
{
 noDebugFlagOnForcedSampling: false,
  throttler: throttler.DefaultThrottler{}, 
.
.
.
```

An even better way would be to use a map, similar to https://github.com/vprithvi/jaeger-client-go/blob/b8197b05185b508c9eb7542b498791d11494fe69/context_test.go#L83 so that we can name the test cases so that we get nicer output when they fail